### ISSUE TITLE
微调qwen2.5vl3b在已经微调了一部分数据后卡住

### ISSUE THREAD

**Issue by sss-lzh** [modelscope/ms-swift#6498]
我在使用ms-swift框架对Qwen2.5-VL-3B-Instruct model进行全量sft微调的时候，出现微调到 838/1298 步时卡住。
表现为gpu显存依旧占着未释放，但显卡功率很低，利用率为0不工作，同时终端输出没有报错信息，就卡在那个进度位置。
用swanlab监控训练信息发现唯一可能相关的变化就是在卡住的时候Process Memory In Use (non-swap) (MB)升高了一些，在网上没有找到相关问题，问ai觉得可能是内存泄露，求大家帮帮忙。训练环境是单卡H100。
卡住时终端是这样的：

> {'loss': 0.8519474, 'grad_norm': 0.22167969, 'learning_rate': 6.54e-06, 'token_acc': 0.76390817, 'epoch': 0.63, 'global_step/max_steps': '820/1298', 'percentage': '63.17%', 'elapsed_time': '4h 3m 5s', 'remaining_time': '2h 21m 42s', 'memory(GiB)': 64.48, 'train_speed(iter/s)': 0.056219}
Train:  63%|��������������������������������������������������������������                                    | 822/1298 [4:03:45<2:31:48, 19.14s/it][h264 @ 0x5743bec0] mmco: unref short failure
[h264 @ 0x5743bec0] mmco: unref short failure
[h264 @ 0x5743bec0] mmco: unref short failure
{'loss': 0.90057859, 'grad_norm': 0.19238281, 'learning_rate': 6.42e-06, 'token_acc': 0.75730448, 'epoch': 0.64, 'global_step/max_steps': '825/1298',{'loss': 0.83218765, 'grad_norm': 0.20214844, 'learning_rate': 6.31e-06, 'token_acc': 0.77009307, 'epoch': 0.64, 'global_step/max_steps': '830/1298', 'percentage': '63.94%', 'elapsed_time': '4h 6m 1s', 'remaining_time': '2h 18m 43s', 'memory(GiB)': 64.48, 'train_speed(iter/s)': 0.056226}18.11s/it]
{'loss': 0.84590149, 'grad_norm': 0.21972656, 'learning_rate': 6.19e-06, 'token_acc': 0.77105664, 'epoch': 0.64, 'global_step/max_steps': '835/1298', 'percentage': '64.33%', 'elapsed_time': '4h 7m 28s', 'remaining_time': '2h 17m 13s', 'memory(GiB)': 64.48, 'train_speed(iter/s)': 0.056236}
Train:  65%|����������������������������������������������������������������                                  | 838/1298 [4:08:14<2:01:21, 15.83s/it]

训练环境是：

> torch                              2.8.0
> transformers                  4.57.1
> qwen-vl-utils                 0.0.11
> ms_swift                      3.9.2
> decord                        0.6.0
> torchvision                   0.23.0

训练配置如下：
```
nproc_per_node=1
CUDA_VISIBLE_DEVICES=0 \
NPROC_PER_NODE=$nproc_per_node \
VIDEO_MAX_PIXELS=1003520 \
FPS_MAX_FRAMES=32 \
PYTORCH_CUDA_ALLOC_CONF="expandable_segments:True" \
swift sft \
    --model /home/hello/weights/Qwen2.5-VL-3B-Instruct \
    --model_type qwen2_5_vl \
    --train_type full \
    --target_modules all-linear \
    --dataset /home/hello/final_dataset/sft_data_fixed.jsonl  \
    --split_dataset_ratio 0.01 \
    --torch_dtype bfloat16 \
    --attn_impl flash_attn \
    --freeze_vit true \
    --freeze_llm true \
    --freeze_aligner false \
    --num_train_epochs 1 \
    --per_device_train_batch_size 4 \
    --per_device_eval_batch_size 4 \
    --gradient_accumulation_steps 4 \
    --gradient_checkpointing true \
    --learning_rate 2e-5 \
    --eval_steps 500 \
    --save_steps 500 \
    --save_total_limit 3 \
    --logging_steps 5 \
    --max_length 8192 \
    --output_dir output \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 0 \
    --report_to swanlab
```

**Comment by Jintao-Huang** [comment]
pip install py-spy
py-spy dump --pid xxx

看看卡在哪了

**Comment by sss-lzh** [comment]
好的，我之前kill了训练进程，等我重新启动一下，运行到卡住的位置再发送我的情况。

**Comment by sss-lzh** [comment]
您好，这是输出的结果，我不知道是什么问题，能帮一下吗？
```
Process 3834885: /home/hello/miniconda3/envs/opengv/bin/python3.10 -u /home/hello/miniconda3/envs/opengv/lib/python3.10/site-packages/swift/cli/sft.py --model /home/hello/weights/Qwen2.5-VL-3B-Instruct --model_type qwen2_5_vl --train_type full --target_modules all-linear --dataset /home/hello/final_dataset/sft_data_fixed.jsonl --split_dataset_ratio 0.01 --torch_dtype bfloat16 --attn_impl flash_attn --freeze_vit true --freeze_llm true --freeze_aligner false --num_train_epochs 1 --per_device_train_batch_size 4 --per_device_eval_batch_size 4 --gradient_accumulation_steps 4 --gradient_checkpointing true --learning_rate 2e-5 --eval_steps 500 --save_steps 500 --save_total_limit 3 --logging_steps 5 --max_length 8192 --output_dir output --warmup_ratio 0.05 --dataloader_num_workers 0 --report_to swanlab
Python v3.10.18 (/home/hello/miniconda3/envs/opengv/bin/python3.10)

Thread 0x7AE3B662D480 (active): "MainThread"
    __call__ (decord/_ffi/_ctypes/function.py:173)
    get_batch (decord/video_reader.py:175)
    _read_video_decord (qwen_vl_utils/vision_process.py:316)
    _new_read_video_decord (swift/llm/model/model/qwen.py:729)
    fetch_video (qwen_vl_utils/vision_process.py:397)
    replace_tag (swift/llm/template/template/qwen.py:310)
    _pre_tokenize (swift/llm/template/base.py:884)
    _simplify_context_list (swift/llm/template/base.py:712)
    _encode (swift/llm/template/base.py:1248)
    _encode (swift/llm/template/template/qwen.py:340)
    _encode_truncated (swift/llm/template/base.py:1201)
    encode (swift/llm/template/base.py:497)
    decorate_context (torch/utils/_contextlib.py:120)
    __getitem__ (swift/llm/dataset/utils.py:97)
    <listcomp> (torch/utils/data/_utils/fetch.py:52)
    fetch (torch/utils/data/_utils/fetch.py:52)
    _next_data (torch/utils/data/dataloader.py:790)
    __next__ (torch/utils/data/dataloader.py:734)
    __iter__ (swift/llm/data_loader.py:80)
    get_batch_samples (transformers/trainer.py:5654)
    _inner_training_loop (transformers/trainer.py:2618)
    train (transformers/trainer.py:2325)
    train (swift/trainers/mixin.py:794)
    train (swift/llm/train/sft.py:243)
    run (swift/llm/train/sft.py:195)
    main (swift/llm/base.py:49)
    sft_main (swift/llm/train/sft.py:340)
    <module> (swift/cli/sft.py:10)
Thread 0x7ADFEDFFF640 (idle): "Thread-1"
    wait (threading.py:324)
    wait (threading.py:607)
    run (tqdm/_monitor.py:60)
    _bootstrap_inner (threading.py:1016)
    _bootstrap (threading.py:973)
Thread 0x7ADF861F8640 (active): "Thread-2 (_read_thread)"
    _recv_msg (torch/_inductor/compile_worker/subproc_pool.py:61)
    _read_thread (torch/_inductor/compile_worker/subproc_pool.py:195)
    run (threading.py:953)
    _bootstrap_inner (threading.py:1016)
    _bootstrap (threading.py:973)
Thread 0x7ADF869F9640 (active): "MsgUploader"
    new_task (swanlab/core_python/uploader/thread/start_thread.py:120)
    run (threading.py:953)
    _bootstrap_inner (threading.py:1016)
    _bootstrap (threading.py:973)
Thread 0x7ADF871FA640 (active): "ThreadPoolExecutor-3_0"
    _worker (concurrent/futures/thread.py:81)
    run (threading.py:953)
    _bootstrap_inner (threading.py:1016)
    _bootstrap (threading.py:973)
Thread 0x7ADF5D7FA640 (idle): "Thread-6"
    wait (threading.py:324)
    wait (threading.py:607)
    run (tqdm/_monitor.py:60)
    _bootstrap_inner (threading.py:1016)
    _bootstrap (threading.py:973)
Thread 0x7ADF56FFD640 (active)
Thread 0x7ADF457FE640 (idle): "Thread-1146"
    wait (threading.py:324)
    wait (threading.py:607)
    run (threading.py:1376)
    _bootstrap_inner (threading.py:1016)
    _bootstrap (threading.py:973)
```

**Comment by scv11** [comment]
deepspeed改成zero2试试

**Comment by sss-lzh** [comment]
@scv11 感谢回复，我将尝试您给出的方法

**Comment by sss-lzh** [comment]
@scv11 试了在835/1298的时候发生了同样的问题，不过最近几次实验基本都是在835或者838这两步的时候发生类似问题，可能是数据原因，但它并没有给出报错信息，之前用AI写了个数据检查的脚本没查出问题，我应该再仔细检查，再次感谢您的回复。
